### PR TITLE
Allow user to specify columns and query expression for subsetting and filtering

### PR DESCRIPTION
**Pull request by chuckwondo**
Allow user to control specific columns to include in output, as well as
query expression for selecting/filtering rows.  This might also allow
the final output file to be small enough to load completely into memory.

Also, refactor subsetting logic to improve performance.

### CODE REVIEW

File: gedi-subset/gedi_utils.py [hunk 1]
```diff
@@ -177,7 +177,7 @@ def spatial_filter(beam, aoi):
 
 @curry
 def subset_h5(
```
Comment (wildintellect) [inline gedi-subset/gedi_utils.py:179]: If we don't use this anymore should we remove?

File: gedi-subset/subset.sh [hunk 2]
```diff
@@ -16,7 +16,19 @@ if test -d input; then
...
+
+    n_actual=${#}
+    n_expected=3
+
+    if test ${n_actual} -gt 0 -a ${n_actual} -ne ${n_expected}; then
+        echo "Expected ${n_expected} inputs, but got ${n_actual}:" $(printf " '%b'" "$@") >&2
+        exit 1
+    fi
+
+    columns=$(test "${1:--}" != "-" && echo " --columns '${1:--}'")
```
Comment (wildintellect) [inline gedi-subset/subset.sh:28]: Do we specify default columns anywhere?

**Review by wildintellect — APPROVED** [review]
Questions:
- How do we plan to handle defaults options for query and columns?
- Should we be using a dev/testing branch to test before merging to main (maybe not critical for a repo like this)?
- also seems we could comment more
- do we have linting/testing?

None of these are blockers for this version.

File: gedi-subset/subset.sh (hunk 2 above)
Comment (chuckwondo) [reply gedi-subset/subset.sh:28]: Defaults are documented in the README.

**Comment by chuckwondo** [comment]
To answer your questions:

- Default input values are described in the README
- I've been thinking about branching as well, and I do think we should create a `develop` branch
- By "comment more" do you mean code doc comments? If so, yes, I agree, we should add more. I started doing so a couple of PRs back, but we need to continue, and also clean up some cruft now that we're tightening things up little by little.
- Regarding linting, isort and black are in the list of deps, but there are not pre-commit hooks, nor and GitHub Actions to apply/enforce such linting. We do not yet have any tests in place, but agree we need to start adding them.

**Pull request merged**